server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -401,32 +400,21 @@ public DocumentMapperForType documentMapperWithAutoCreate() {
 
     /**
      * Given the full name of a field, returns its {@link MappedFieldType}.
+     * @deprecated Use {@link #snapshot()} and {@link Snapshot#fieldType(String)}
      */
+    @Deprecated
```

Comment:
These are still used in a bunch of places. I *think* we should avoid them if we decide that this is the right way to go.